movement/watch_faces/settings/finetune_face.c

```diff
@@ -197,7 +198,14 @@ bool finetune_face_loop(movement_event_t event, movement_settings_t *settings, v
                 finetune_adjust_subseconds(250);
             } else if (finetune_page == 2 && finetune_get_hours_passed() >= 6) {
                 // Applying ppm correction, only if >6 hours passed
-                nanosec_state.freq_correction += (int)round(finetune_get_correction() * 100);
+                int32_t f = nanosec_state.freq_correction + (int)round(finetune_get_correction() * 100);
```

Comment:
I think a cleaner approach is something like:

```c
int correction = (int)round(finetune_get_correction() * 100);
if (nanosec_state.freq_correction > 0 && correction > SHRT_MAX - nanosec_state.freq_correction) {
    nanosec_state.freq_correction = SHRT_MAX;
} else if (nanosec_state.freq_correction < 0 && correction < SHRT_MIN - nanosec_state.freq_correction) {
    nanosec_state.freq_correction = SHRT_MIN;
} else {
    nanosec_state.freq_correction += correction;
}
```

since it avoids the need for a intermediate variable. Does that look right to you?